Bluewire.Tools.Runner/GenerateScripts/ToolRunner.cs

```diff
@@ -0,0 +1,183 @@
+            {
+                {"d=|directory=", "Produce scripts in the specified directory. Default: same directory as this runner.", o => arguments.TargetDirectory = o },
+                {"r=|runner=", "Specify the path to the runner, relative to the directory which will contain the scripts.", o => arguments.RunnerPath = o }
+            };
+        }
+
+        public int RunMain(string[] args, string parentArgs)
+        {
+            var arguments = new Arguments();
+            var options = CreateOptions(arguments);
+            var sessionArgs = new SessionArguments<Arguments>(arguments, options);
+            sessionArgs.Application += parentArgs;
+            return new ConsoleSession<Arguments>(sessionArgs).Run(args, async a => await new Impl(a, toolNames).Run());
+        }
+
+        class Impl
+        {
+            private readonly Arguments arguments;
+            private readonly string[] toolNames;
+
+            public Impl(Arguments arguments, string[] toolNames)
+            {
+                this.arguments = arguments;
+                this.toolNames = toolNames;
+
+                Log.Configure();
+                Log.SetConsoleVerbosity(arguments.Verbosity);
+            }
+
+            public Task<int> Run()
+            {
+                var targetDirectory = Path.GetFullPath(arguments.TargetDirectory);
+                Log.Console.Debug($"Scripts will be produced in: {targetDirectory}");
+
+                var runnerInvocation = GetRunnerInvocation(targetDirectory, arguments.RunnerPath);
+                var shellRunnerInvocation = ToLinuxPath(runnerInvocation);
+                Log.Console.Debug($"Using Linux runner invocation: {shellRunnerInvocation}");
+
+                if (!Directory.Exists(targetDirectory))
+                {
+                    Log.Console.Debug("Target directory does not exist. It will be created.");
+                    Directory.CreateDirectory(targetDirectory);
+                }
+
+                foreach (var toolName in toolNames)
+                {
+                    var nameRoot = Path.Combine(targetDirectory, toolName);
+                    Log.Console.Debug($"Writing {nameRoot}...");
+                    File.WriteAllText(nameRoot, GetShellScriptContent(shellRunnerInvocation, toolName));
+                    Log.Console.Debug($"Writing {nameRoot}.cmd...");
+                    File.WriteAllText($"{nameRoot}.cmd", GetCommandFileContent(runnerInvocation, toolName));
+                }
+                return Task.FromResult(0);
+            }
+
+            private string GetShellScriptContent(string shellRunnerInvocation, string toolName)
+            {
+                return $@"#!/bin/sh
+{shellRunnerInvocation} --tool {toolName} ""$@""
+";
+            }
+            private string GetCommandFileContent(string runnerInvocation, string toolName)
+            {
+                return $@"@echo off
+{runnerInvocation} --tool {toolName} %*
+";

```

Comment:
This is a GitHub rendering artifact. Examining the file shows that all line breaks are CRLF.